pkg/cloud/services/ec2/securitygroups.go

```diff
@@ -62,9 +62,9 @@ func (s *Service) reconcileSecurityGroups() error {
 	// Declare all security group roles that the reconcile loop takes care of.
 	roles := []infrav1.SecurityGroupRole{
 		infrav1.SecurityGroupBastion,
+		infrav1.SecurityGroupLB,
```

Comment:
This change is to make the test actually catch the scope "leakage," since the LB is the only security group (or, really, the only place in CAPA) that we add a conditional tag.